iModelCore/ECPresentation/Source/Content/Content.cpp

```diff
@@ -1038,7 +1039,11 @@ bvector<InstanceLabelOverrideCP> ContentDescriptor::DisplayLabelField::CloneLabe
 +---------------+---------------+---------------+---------------+---------------+------*/
 ContentDescriptor::Field::TypeDescriptionPtr ContentDescriptor::CalculatedPropertyField::_CreateTypeDescription() const
     {
-    return new PrimitiveTypeDescription("string");
+    if (m_type == PRIMITIVETYPE_Point3d || m_type == PRIMITIVETYPE_Point2d ||
+        m_type == PRIMITIVETYPE_Binary || m_type == PRIMITIVETYPE_IGeometry)
+        return new PrimitiveTypeDescription(EC_PRIMITIVE_TYPENAME_STRING);

```

Comment:
We know this should not happen, right? So I'd say we should have a `switch` statement that names all supported types and returns a type description for them. And then by default we should just throw, saying that calculated properties field doesn't support that type.

Reply:
Yes, this shouldn't happen. Changed to your suggestion.